### ISSUE TITLE
NonZero* support

### ISSUE THREAD

**Issue by passcod** [sharksforarms/deku#137]
It would be nice to have `NonZero*` support, and/or `Option<NonZero*>`, built-in.

**Comment by sharksforarms** [comment]
Hey @passcod if you have the time, please give https://github.com/sharksforarms/deku/pull/140 a try

**Comment by sharksforarms** [comment]
For `Option<NonZero*>` it may not be possible (you'll need a `cond` attribute still), waiting on specialization to do this. (For example, specialize over Option<NonZeroU8> and also have impl over Option<T>)
